app.py

```diff
@@ -58,34 +137,66 @@ def index():
 # will have to be updated
 @app.route("/registration", methods=["GET", "POST"])
 def registration():
-    """
-    The registration page of the app. Allows new users to
-    register for an account.
-    """
-
-    username = flask.request.form.get("username")
-
-    if username:
-        return flask.redirect(flask.url_for("index"))
-
-    return flask.render_template("registration.html")
+    form = RegisterForm()
+
+    # checks the input to see if the username and password are valid
+    if form.username.data == None:
+        flash("")
+    elif len(form.username.data) < 4:
+        flash("Please enter a username that has more than 4 characters")
+    elif len(form.username.data) > 20:
+        flash("Please enter a username that is less than 20 characters")
+    elif len(form.password.data) < 4:
+        flash("Please enter a password greater than 4 characters")
+    elif len(form.password.data) > 20:
+        flash("Please enter a password shorter than 20 characters")
+
+    # hashes password and stores it in the db
+    if form.validate_on_submit():
+        pw_hash = bcrypt.generate_password_hash(form.password.data).decode("utf-8")
+        newUser = BTUser(username=form.username.data, password_hash=pw_hash)
+        db.session.add(newUser)
+        db.session.commit()
+        print("redirected")
+        flash("Account Created!")
+        return redirect(url_for("login"))
+
+    return render_template("registration.html", form=form)
 
 
 # NOTE user cannot be verified until DB is set up and conditional statements
 # will have to be updated
```

Comment:
These comments (and the similar comments above `registration`) can be removed now.